modules/apps/29-fee/keeper/msg_server_test.go

```diff
@@ -21,90 +25,92 @@ func (suite *KeeperTestSuite) TestRegisterPayee() {
 
 	testCases := []struct {
 		name     string
-		expPass  bool
 		malleate func()
+		expErr   error
 	}{
 		{
 			"success",
-			true,
 			func() {},
+			nil,
 		},
 		{
 			"channel does not exist",
-			false,
 			func() {
 				msg.ChannelId = "channel-100" //nolint:goconst
 			},
+			channeltypes.ErrChannelNotFound,
 		},
 		{
 			"channel is not fee enabled",
-			false,
 			func() {
 				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
 			},
+			types.ErrFeeNotEnabled,
 		},
 		{
 			"given payee is not an sdk address",
-			false,
 			func() {
 				msg.Payee = "invalid-addr"
 			},
+			errors.New("decoding bech32 failed: invalid separator index -1"),
 		},
 		{
 			"payee is a blocked address",
-			false,
 			func() {
 				msg.Payee = suite.chainA.GetSimApp().AccountKeeper.GetModuleAddress(transfertypes.ModuleName).String()
 			},
+			ibcerrors.ErrUnauthorized,
 		},
 	}
 
 	for _, tc := range testCases {
 		tc := tc
 
-		suite.SetupTest()
-		suite.path.Setup()
+		suite.Run(tc.name, func() {
```

Comment:
There are a few instances in this PR where I introduced `t.Run` or `suite.Run`. It helped me debug things but I also think it's better to have it this way, even though it increases the diff. Happy to revert if people find it would be better suited for another PR